Tasks/Common/ios-signing-common/ios-signing-common.ts

```diff
@@ -0,0 +1,209 @@
+import fs = require('fs');
+import path = require('path');
+import Q = require('q');
+import tl = require('vsts-task-lib/task');
+import {ToolRunner} from 'vsts-task-lib/toolrunner';
+
+var userProvisioningProfilesPath = path.join(tl.getVariable('HOME'), 'Library', 'MobileDevice', 'Provisioning Profiles');
+
+/**
+ * Creates a temporary keychain and installs the P12 cert in the temporary keychain
+ * @param keychainPath, the path to the keychain file
+ * @param keychainPwd, the password to use for unlocking the keychain
+ * @param p12CertPath, the P12 cert to be installed in the keychain
+ * @param p12Pwd, the password for the P12 cert
+ */
+export async function installCertInTemporaryKeychain(keychainPath : string, keychainPwd: string, p12CertPath : string, p12Pwd: string) {
+
+    //delete keychain if exists
+    await deleteKeychain(keychainPath);
+
+    //create keychain
+    var createKeychainCommand : ToolRunner =  tl.createToolRunner(tl.which('security', true));
```

Comment:
tl has an execsync function for quick CLIs like this.  It doesn't stream output to the live console but that's not needed here.  It's one line of readable code as opposed to this.